framework/test/mocha/unit/modules/chain/logic/bft.js

```diff
@@ -0,0 +1,323 @@
+				const header1 = blockHeaderFixture({ height: 10 });
+				const header2 = blockHeaderFixture({ height: 11 });
+				const header3 = blockHeaderFixture({ height: 13 });
+
+				list.add(header1).add(header2);
+
+				expect(() => {
+					list.add(header3);
+				}).to.throw(
+					'Block header with height 12 or 9 can be added at the moment'
+				);
+			});
+
+			it('should throw error if block header added is not one step less than first item', async () => {
+				const header1 = blockHeaderFixture({ height: 10 });
+				const header2 = blockHeaderFixture({ height: 9 });
+				const header3 = blockHeaderFixture({ height: 7 });
+
+				list.add(header1).add(header2);
+
+				expect(() => {
+					list.add(header3);
+				}).to.throw(
+					'Block header with height 11 or 8 can be added at the moment'
+				);
+			});
+
+			it('should keep headers pushed in sequence', async () => {
+				const header1 = blockHeaderFixture({ height: 10 });
+				const header2 = blockHeaderFixture({ height: 11 });
+				const header3 = blockHeaderFixture({ height: 12 });
+
+				list.add(header1);
+				list.add(header2);
+				list.add(header3);
+
+				expect(list.items).to.eql([header1, header2, header3]);
+			});
+
+			it('should remove the first header if list size increased', async () => {
+				const header1 = blockHeaderFixture({ height: 1 });
+				const header2 = blockHeaderFixture({ height: 2 });
+				const header3 = blockHeaderFixture({ height: 3 });
+				const header4 = blockHeaderFixture({ height: 4 });
+				const header5 = blockHeaderFixture({ height: 5 });
+				const header6 = blockHeaderFixture({ height: 6 });
+
+				list
+					.add(header1)
+					.add(header2)
+					.add(header3)
+					.add(header4)
+					.add(header5)
+					.add(header6);
+
+				expect(list.items).to.be.eql([
+					header2,
+					header3,
+					header4,
+					header5,
+					header6,
+				]);
+			});
+		});
+
+		describe('remove()', () => {
+			let header1;
+			let header2;
+			let header3;
+			let header4;
+			let header5;
+
+			beforeEach(async () => {
+				header1 = blockHeaderFixture({ height: 1 });
+				header2 = blockHeaderFixture({ height: 2 });
+				header3 = blockHeaderFixture({ height: 3 });
+				header4 = blockHeaderFixture({ height: 4 });
+				header5 = blockHeaderFixture({ height: 5 });
+
+				list
+					.add(header1)
+					.add(header2)
+					.add(header3)
+					.add(header4)
+					.add(header5);
+
+				expect(list.items).to.be.eql([
+					header1,
+					header2,
+					header3,
+					header4,
+					header5,
+				]);
+			});
+
+			it('should remove last item from the list if passed without height', async () => {
+				list.remove();
+				expect(list.items).to.be.eql([header1, header2, header3, header4]);
+			});
+			it('should remove all items upto and including the provided height', async () => {
+				list.remove({ beforeHeight: 2 });
+				expect(list.items).to.be.eql([header1, header2]);
```

Comment:
This expectation does not match the test description

Reply:
It matches what ever height we pass in `list.remove()` it should remove all blocks above that height. The list contains 5 blocks, we asked to remove blocks above height 2 and so we have `[header1, header2]` left in the list.

Reply:
The test description states that it removes the headers including the provided height. 
`should remove all items upto and including the provided height`